src/service/viewport-impl.js

```diff
@@ -1093,8 +1097,11 @@ export class ViewportBindingNatural_ {
 
   /** @override */
   getScrollTop() {
-    return this.getScrollingElement_()./*OK*/scrollTop ||
+    const pageScrollTop = this.getScrollingElement_()./*OK*/scrollTop ||
         this.win./*OK*/pageYOffset;
+    const host = this.ampdoc.getRootNode().host;
+    const hostOffset = (host && host.offsetTop) || 0;
+    return pageScrollTop - hostOffset;
```

Comment:
Is there a better way to apply this offset? The value returned here could also be expressed `-this.getLayoutRect(this.ampdoc.getBody(), 0, 0);`, but that is probably much slower.

Also, do the iOS embed binding classes also need to account for the offset of the shadow root inside the host page?